enhancements/baremetal-config.md

```diff
@@ -0,0 +1,153 @@
+---
+title: Baremetal IPI Config Enhancement
+authors:
+  - "@sadasu"
+reviewers:
+  - "@deads2k"
+  - "@hardys"
+  - "@enxebre"
+  - "@squeed"
+  - "@abhinavdahiya"
+
+approvers:
+  - "@deads2k"
+
+creation-date: 2019-10-23
+last-updated: 2019-10-23
+status: implemented
+see-also:
+  - "https://github.com/openshift/api/pull/480"  
+replaces:
+  - 
+superseded-by:
+  - 
+---
+
+# Config required for Baremetal IPI deployments
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+None.
+
+## Summary
+
+The configuration required to bring up a Baremetal IPI (metal3) cluster needs
+to be availabe in a Config Resource. The installer aleady adds some configuration
+to the Infrastructure Config CR. It is also aware of the additional config
+required during the deployment of a metal3 cluster. This enhancement proposes
+to extend the BareMetaPlatformStatus CR to hold these additional values needed
+for a metal3 deployment.
+
+## Motivation
+
+Adding these additional config items to the BareMetalPlatformStatus CR, allows
+versioning of these config items.
+
+### Goals
+
+The goal of this enhancement is to enable the installer to add vital config
+values for a Baremetal IPI deployment to a CR. The MAO will then be able to
+access these values when it kicks off a metal3 cluster.
+
+Update lifecycle for bootimage is an unsolved problem for all platforms. For
+more information, refer to : https://github.com/openshift/os/issues/381.
+
+### Non-Goals
+
+All config values except the RHCOSImageUrl are not expected to change after a
+metal3 cluster is up. Handling changes to this value after install time has
+not been explored and that is a non-goal at this time.
+
+### Proposal
+
+To support configuration of a Baremetal IPI (metal3) deployment, a set of new
+config items are required to be made availabe via a Config Resource. The
+BareMetalPlatformStatus is an ideal fit because it contains configuration
+that is specific to a Baremetal deployment.
+
+This proposal aims to add the following to BareMetalPlatformStatus:
+
+1. ProvisioningInterface : This is the interface name on the underlying
+baremetal server which would be physically connected to the provisioning
+network.
+
+2. ProvisioningIP : This is the IP address used to to bring up a NIC on
```

Comment:
hmm, this is a little difficult to understand what exactly this is, I don't completely understand baremetal-server provisioinig... a little bit context here?

Reply:
This is the IP address that the interface above is configured with - we've got a mechanism that statically configures ProvisioningInterface on whatever master node has the metal3 baremetal-operator pod running, but this is a short lease such that if/when the pod gets rescheduled the IP goes down on the old host, and is the brought back up on the new host.

Reply:
> This is the IP address that the interface above is configured with - we've got a mechanism that statically configures ProvisioningInterface on whatever master node has the metal3 baremetal-operator pod running, but this is a short lease such that if/when the pod gets rescheduled the IP goes down on the old host, and is the brought back up on the new host.

@sadasu this would be good info to include in the body of the doc. Maybe link to any metal3 docs about it, too?

Reply:
> this would be good info to include in the body of the doc. Maybe link to any metal3 docs about it, too?

A number of the "what is this?" threads on this PR could benefit from a link to or local docs about how metal3 bootstrapping/management works.  Currently we have bits and pieces of it described in response comments in the PR; would be nice to collect that up and put an explainer in a _Background_ section right before this _Proposal_ section.

Reply:
Creating a background "enhancement" here https://github.com/openshift/enhancements/pull/102 so that we can point to that from this enhancement request and others that follow. Feel free to comment on that PR too.

As per the actual config items, this enhancement request would be the one where all the details would exist.